src/uu/rm/src/rm.rs

```diff
@@ -514,22 +514,18 @@ fn prompt_file(path: &Path, options: &Options) -> bool {
                     prompt_yes!("remove file {}?", path.quote())
                 };
             }
-            prompt_file_permission_readonly(path, Ok(metadata))
         }
         Err(err) => {
             if err.kind() != ErrorKind::PermissionDenied {
                 return true;
             }
-            prompt_file_permission_readonly(path, fs::metadata(path))
         }
     }
+    prompt_file_permission_readonly(path)
```

Comment:
Good catch, I didn't notice this simplification :)

Reply:
Should I still leave it in its own function or move it into prompt_file?

Reply:
It's fine as it is.